Libraries/dotNetRdf.Core/Writing/Formatting/NTriplesFormatter.cs

```diff
@@ -271,7 +277,19 @@ public override string FormatUri(Uri u)
             //    return FormatUri(u.ToString());
             //}
 
-            return FormatUri(Rfc3987Formatter.EscapeUriString(u.ToString()));
+            var uriString = u.ToString();
+            if (uriString.EndsWith("/") && !u.OriginalString.EndsWith("/"))
+                uriString = uriString.Substring(0, uriString.Length - 1);
+            else if (!uriString.EndsWith("/") && u.OriginalString.EndsWith("/"))
+                uriString += '/';
+            else if (uriString.Contains("/#") && !u.OriginalString.Contains("/#"))
+                uriString = uriString.Replace("/#", "#");
+            else if (uriString.Contains("/#") && !u.OriginalString.Contains("/#"))
```

Comment:
This looks like a duplicate of lines 285-286

Reply:
Oh, oops, pasted that block twice ^^;

Should be fixed.